src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/validation/RecallValidationService.kt

```diff
@@ -39,6 +41,34 @@ class RecallValidationService(
     }
   }
 
+  internal fun validateRemandPeriodsAgainstSentenceDates(sourceData: PrisonApiSourceData): List<ValidationMessage> {
+    val remandPeriods = sourceData.bookingAndSentenceAdjustments.sentenceAdjustments
+      .filter { it.type == SentenceAdjustmentType.REMAND }
+
+    val validationMessages = mutableListOf<ValidationMessage>()
+
+    sourceData.sentenceAndOffences.forEach { sentence ->
```

Comment:
Would it make more sense to loop through the remand periods and then match them to a sentence, rather than the other way around?

Reply:
I've implmented that change